### PR TITLE
feat(crypt): use ed25519 algorithm for signing

### PR DESCRIPTION
**Pull request by henri-hulski**

### CODE REVIEW

File: packages/apps/vasudeva/src/controller/UserController.ts [hunk 1]
```diff
@@ -96,7 +95,7 @@ export class UserController {
     })
     await User.save(user)
 
-    const collectionId = idWithPrefix(userId, 'o')
+    const collectionId = crypt.idAddPrefix(userId, 'p')
```
Comment (gaspard) [inline packages/apps/vasudeva/src/controller/UserController.ts]: Could we have functions for each of these prefix handling operations ? I think it is not so good to have literals like 'p' and it makes the code more readable:

```ts
const collectionId = crypt.privateCollectionId(userId)
```

Comment (henri-hulski) [reply packages/apps/vasudeva/src/controller/UserController.ts]: Should be fixed now!

File: packages/node_modules/@lucidogen/crypt/src/makeId.test.ts [hunk 2]
```diff
@@ -8,84 +8,110 @@ import {
...
+    const userId = crypt.userId()
+    const id = crypt.workCollectionId(userId)
+    expect(base64ToBuffer(id.slice(2)).length).toBe(32)
+    expect(id.slice(0, 2)).toBe('w.')
+  })
+
+  it('should generate a work collection id from private collection id', () => {
+    const userId = crypt.userId()
+    const privateCollectionId = crypt.privateCollectionId(userId)
+    const id = crypt.privateCollectionId(privateCollectionId)
```
Comment (gaspard) [inline packages/node_modules/@lucidogen/crypt/src/makeId.test.ts]: Should be `workCollectionId` but we should really just extract `userId` and not do this directly. If we are in a situation to find the `workCollectionId` we have the `userId` not another collection.

File: packages/node_modules/@lucidogen/crypt/src/makeId.test.ts [hunk 3]
```diff
@@ -8,84 +8,110 @@ import {
...
-})
 
-describe('idPrefix', () => {
-  it('should get prefix from id', () => {
-    const id = makeId('x')
-    expect(idPrefix(id)).toBe('x')
+  it('should generate a private collection id from work collection id', () => {
+    const userId = crypt.userId()
+    const workCollectionId = crypt.workCollectionId(userId)
+    const id = crypt.privateCollectionId(workCollectionId)
```
Comment (gaspard) [inline packages/node_modules/@lucidogen/crypt/src/makeId.test.ts]: Please, no direct extracting: it is confusing and adds unnecessary complexity. We just need a way to get back to `userId`.

There is no reason to convert private to work. The only thing we need is user<->work and user<->private.

File: packages/node_modules/@lucidogen/crypt/src/makeId.ts [hunk 4]
```diff
@@ -1,49 +1,60 @@
...
+}
+
+export function workCollectionId(userId: string): string {
+  return mutateIdPrefix(userId, 'w')
+}
+
+export function userId(id?: string): string {
+  if (id) {
+    if (id.slice(0, 2) !== 'p.' && id.slice(0, 2) !== 'w.') {
+      throw new Error('UserId can only be extracted from privateCollectionId or workCollectionId.')
```
Comment (gaspard) [inline packages/node_modules/@lucidogen/crypt/src/makeId.ts]: This guard is good.

File: packages/node_modules/@lucidogen/crypt/src/makeId.test.ts (hunk 3 above)
Comment (henri-hulski) [reply packages/node_modules/@lucidogen/crypt/src/makeId.test.ts]: Actually this happened more like a side effect. The internal `mutateIdPrefix(id, prefix)` allowed removing, replacing and adding a prefix.
Fixed now.

**Pull request merged**